ipn/ipnlocal/profiles.go

```diff
@@ -124,37 +141,45 @@ func init() {
 // provided prefs, which may be accessed via CurrentPrefs.
 func (pm *profileManager) SetPrefs(prefsIn ipn.PrefsView) error {
 	prefs := prefsIn.AsStruct().View()
-	ps := prefs.Persist()
-	if ps == nil || ps.LoginName == "" {
+	newPersist := prefs.Persist()
+	if newPersist == nil || newPersist.LoginName == "" {
 		return pm.setPrefsLocked(prefs)
 	}
-	up := ps.UserProfile
+	up := newPersist.UserProfile
 	if up.LoginName == "" {
-		up.LoginName = ps.LoginName
+		// Backwards compatibility with old prefs files.
+		up.LoginName = newPersist.LoginName
+	} else if newPersist.LoginName != "" && newPersist.LoginName != up.LoginName {
```

Comment:
`newPersist.LoginName != ""` should always be true given the early return at line 145.

Reply:
Done.